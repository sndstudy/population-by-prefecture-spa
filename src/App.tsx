import { QueryClient, QueryClientProvider } from 'react-query';
import Main from './container/Main';

const queryClient = new QueryClient();

function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <Main />
    </QueryClientProvider>
  );
}

export default App;
