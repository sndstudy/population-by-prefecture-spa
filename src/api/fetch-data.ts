import axios from 'axios';

const RESAS_API_ENDPOINT = 'https://opendata.resas-portal.go.jp/';
const API_KEY = process.env['REACT_APP_RESAS_API_KEY'] || '';

const instance = axios.create({
  baseURL: RESAS_API_ENDPOINT,
  timeout: 1000,
  headers: { 'X-API-KEY': API_KEY },
});

export type PrefecturesResponse = {
  message: string;
  result: Prefectures[];
};

export type Prefectures = {
  prefCode: string;
  prefName: string;
};

export type PopulationResponse = {
  message: string;
  result: Population;
};

export type Population = {
  boundaryYear: number;
  data: {
    label: string;
    data: {
      year: number;
      value: number;
      rate?: number;
    }[];
  }[];
};

export const fetchPrefectures = async () => {
  const { data } = await instance.get<PrefecturesResponse>(
    'api/v1/prefectures',
  );
  return data;
};

export const fetchPopulationData = async ({
  queryKey,
}: {
  queryKey: string[];
}) => {
  const [, prefCode] = queryKey;

  const { data } = await instance.get<PopulationResponse>(
    'api/v1/population/composition/perYear',
    {
      params: {
        prefCode,
      },
    },
  );
  return data;
};
