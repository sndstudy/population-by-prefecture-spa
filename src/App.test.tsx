import { render, screen, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { PrefecturesResponse } from './api/fetch-data';
import App from './App';

const server = setupServer(
  rest.get(
    'https://opendata.resas-portal.go.jp/api/v1/prefectures',
    (req, res, ctx) => {
      return res(
        ctx.delay(500),
        ctx.json<PrefecturesResponse>({
          message: 'テストメッセージ',
          result: [
            { prefCode: '1', prefName: '北海道' },
            { prefCode: '2', prefName: '青森県' },
            { prefCode: '3', prefName: '岩手県' },
          ],
        }),
      );
    },
  ),
);

describe('Mainのテスト', () => {
  beforeAll(() => {
    server.listen();
  });
  afterAll(() => {
    server.close();
  });

  it('初期描画', async () => {
    const { asFragment } = render(<App />);

    await waitFor(() => {
      const checkBoxList = screen.getAllByRole('checkbox');
      expect(checkBoxList.length).toBe(3);
      checkBoxList.forEach((v) => {
        // チェックが付いていないことを確認
        expect(v).not.toBeChecked();
      });
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
