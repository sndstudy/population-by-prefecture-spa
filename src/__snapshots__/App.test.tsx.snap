// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainのテスト 初期描画 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    都道府県別 総人口
  </div>
  <div
    class="wrapper"
  >
    <div
      class="prefectures_area"
    >
      <div
        class="section_title"
      >
        都道府県選択
      </div>
      <div>
        <label>
          <input
            type="checkbox"
            value="1"
          />
          北海道
        </label>
      </div>
      <div>
        <label>
          <input
            type="checkbox"
            value="2"
          />
          青森県
        </label>
      </div>
      <div>
        <label>
          <input
            type="checkbox"
            value="3"
          />
          岩手県
        </label>
      </div>
    </div>
    <div
      class="graph_area"
    >
      <div
        data-highcharts-chart="0"
        style="height: 100%; overflow: hidden;"
      >
        <div
          class="highcharts-container "
          dir="ltr"
          id="highcharts-mfmp91h-0"
          style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
        >
          <svg
            aria-label="都道府県別 総人口数"
            class="highcharts-root"
            height="400"
            role="img"
            style="font-family: \\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif; font-size: 12px;"
            version="1.1"
            viewBox="0 0 600 400"
            width="600"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc>
              Created with Highcharts 10.3.3
            </desc>
            <defs>
              <clippath
                id="highcharts-mfmp91h-1-"
              >
                <rect
                  fill="none"
                  height="250"
                  width="580"
                  x="0"
                  y="0"
                />
              </clippath>
            </defs>
            <rect
              class="highcharts-background"
              fill="#ffffff"
              height="400"
              rx="0"
              ry="0"
              width="600"
              x="0"
              y="0"
            />
            <rect
              class="highcharts-plot-background"
              fill="none"
              height="250"
              width="580"
              x="10"
              y="50"
            />
            <rect
              class="highcharts-plot-border"
              data-z-index="1"
              fill="none"
              height="250"
              stroke="#cccccc"
              stroke-width="0"
              width="580"
              x="10"
              y="50"
            />
            <g
              class="highcharts-series-group"
              data-z-index="3"
            />
            <text
              class="highcharts-title"
              data-z-index="4"
              style="color: rgb(51, 51, 51); font-size: 18px; fill: #333333;"
              text-anchor="middle"
              x="300"
              y="64"
            >
              都道府県別 総人口数
            </text>
            <text
              class="highcharts-subtitle"
              data-z-index="4"
              style="color: rgb(102, 102, 102); fill: #666666;"
              text-anchor="middle"
              x="300"
              y="50"
            />
            <text
              class="highcharts-caption"
              data-z-index="4"
              style="color: rgb(102, 102, 102); fill: #666666;"
              text-anchor="start"
              x="10"
              y="312"
            />
            <g
              class="highcharts-legend highcharts-no-tooltip"
              data-z-index="7"
              visibility="hidden"
            >
              <rect
                class="highcharts-legend-box"
                fill="none"
                height="8"
                rx="0"
                ry="0"
                stroke="#999999"
                stroke-width="0"
                width="8"
                x="0"
                y="0"
              />
              <g
                data-z-index="1"
              >
                <g />
              </g>
            </g>
            <text
              class="highcharts-credits"
              data-z-index="8"
              style="cursor: pointer; color: rgb(153, 153, 153); font-size: 9px; fill: #999999;"
              text-anchor="end"
              x="590"
              y="395"
            >
              Highcharts.com
            </text>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
